astropy/convolution/utils.py

```diff
@@ -286,3 +286,48 @@ def discretize_integrate_2D(model, x_range, y_range):
             values[j, i] = dblquad(model, x[i], x[i + 1],
                                    lambda x: y[j], lambda x: y[j + 1])[0]
     return values
+
+def profile_memory_usage_convolve_fft():
+    """
+    Utility to profile the memory usage of convolve_fft
+
+    Requires memory_profiler and timeit
+    """
+    from memory_profiler import memory_usage
+    import timeit
+    from astropy import units as u
+    import time
+    for size in [256,300,512,600,1000,1024,2048]:
+        for kernelsize in (int(size/8), int(size/4), int(size/2), int(size)):
+            for psf_pad in (True,False):
+                for fft_pad in (True,False):
+                    x=np.ones([size,size])
+                    y=np.ones([kernelsize,kernelsize])
+                    t0 = time.time()
+                    # this checks the memory usage many times over the course of the function call
+                    memuse = memory_usage((convolve_fft, (x,y,),
+                                           dict(psf_pad=psf_pad, fft_pad=fft_pad)),
+                                          interval=0.01)
+                    if time.time() - t0 < 1:
+                        # for tests that go fast, repeat them a few times for robustness' sake
+                        number = 10
+                    else:
+                        # for tests that go slow, do them once - we're accurate enough already
+                        number = 1
+                    timeuse = timeit.repeat('convolve_fft(x,y,psf_pad={psf_pad},'
+                                            ' fft_pad={fft_pad})'.format(psf_pad=psf_pad, fft_pad=fft_pad),
+                                            setup=('from astropy.convolution import convolve_fft; import numpy as np;'
+                                                   'x=np.ones([{size},{size}]);'
+                                                   'y=np.ones([{kernelsize},{kernelsize}]);'
+                                                   .format(size=size, kernelsize=kernelsize)),
+                                            repeat=3, number=number)
+                    print("psf_pad={psf_pad:5s}, fft_pad={fft_pad:5s}, size={size:5d}  kernelsize={kernelsize:5d}"
+                          " Mem usage max={0:8.1f}.  Array, kernel size are {1:5.1f} and {2:5.1f}."
+                          " Time use was {meantime:6.3f}"
+                          .format((max(memuse)-min(memuse))*u.MB,
+                                  (x.nbytes*u.byte).to(u.MB),
+                                  (y.nbytes*u.byte).to(u.MB), psf_pad=str(psf_pad),
+                                  fft_pad=str(fft_pad), size=size,
+                                  meantime=np.mean(timeuse)*u.s,
+                                  kernelsize=kernelsize,
+                                 ))
```

Comment:
Maybe break out the contents of the inner loop to a separate function (even defined inline in the main function is fine).